src/ext.js

```diff
@@ -97,230 +99,349 @@ const textAlignItems = {
+                component: 'switch',
+                options: [
+                  {
+                    value: true,
+                    translation: 'Common.Auto',
+                  },
+                  {
+                    value: false,
+                    translation: 'Common.Custom',
+                  },
+                ],
+                defaultValue: true,
               },
-            },
-            fontColor: {
-              show: true,
-              ref: 'content.fontColor',
-              translation: 'ThemeStyleEditor.style.cellFontColor',
-              type: 'object',
-              component: 'color-picker',
-              dualOutput: true,
-            },
-            hoverEffect: {
-              show: true,
-              ref: 'content.hoverEffect',
-              translation: 'ThemeStyleEditor.style.hoverEffect',
-              type: 'boolean',
-              component: 'switch',
-              options: [
-                {
-                  value: true,
-                  translation: 'properties.on',
-                },
-                {
-                  value: false,
-                  translation: 'properties.off',
+              {
+                ref: 'totals.position',
+                translation: 'Common.Position',
+                type: 'string',
+                component: 'dropdown',
+                options: [
+                  {
+                    value: 'noTotals',
+                    translation: 'Common.None',
+                  },
+                  {
+                    value: 'top',
+                    translation: 'Common.Top',
+                  },
+                  {
+                    value: 'bottom',
+                    translation: 'Common.Bottom',
+                  },
+                ],
+                defaultValue: 'noTotals',
+                show(data) {
+                  return !data.totals.show;
                 },
-              ],
-            },
-            hoverColor: {
-              show: (data) => !!data.content.hoverEffect,
-              ref: 'content.hoverColor',
-              translation: 'ThemeStyleEditor.style.hoverStyle',
-              type: 'object',
-              component: 'color-picker',
-              dualOutput: true,
-            },
-            hoverFontColor: {
-              show: (data) => !!data.content.hoverEffect,
-              ref: 'content.hoverFontColor',
-              translation: 'ThemeStyleEditor.style.hoverFontStyle',
-              type: 'object',
-              component: 'color-picker',
-              dualOutput: true,
-            },
+              },
+              {
+                ref: 'totals.label',
+                translation: 'properties.totals.label',
+                type: 'string',
+                expression: 'optional',
+                defaultValue: 'Totals',
+              },
+            ],
+            show: !env?.anything?.sense?.isUnsupportedFeature?.('totals'),
```

Comment:
I refactored the current file because it was hard to follow the order of properties, to make it more readable but my changes are basically focused only on `totals-show `( lines 234 - 282) and `totals aggregation function` ( lines 355- 413).

Reply:
Try to keep big refactors like this out of feature PRs. Plus this adds three issues on codeclimate. This could still be a good idea but lets keep it in a seperate PR

Reply:
Sure! Agree!  It's a quite big refactoring so I'll revert it to the original orders as soon as I finish with the other comments.